src/tree/v2/azure/registerAzureTree.ts

```diff
@@ -20,7 +20,11 @@ interface RegisterAzureTreeOptions {
     refreshEvent: vscode.Event<void>,
 }
 
-export function registerAzureTree(context: vscode.ExtensionContext, options: RegisterAzureTreeOptions): void {
+interface RegisterAzureTreeResult {
+    azureResourceTreeDataProvider: AzureResourceTreeDataProvider;
+}
+
+export function registerAzureTree(context: vscode.ExtensionContext, options: RegisterAzureTreeOptions): RegisterAzureTreeResult {
```

Comment:
Nit: as the type is not used by callers (e.g. it's not exported) and unlikely to require returning other properties, I might dispense with the interface and just return the TDP directly.  I think it's a reasonable thing for callers to expect to be returned from such a function.  (And same comment below.)

Reply:
Agreed, it felt a little strange to me.